### PR TITLE
getDevice/getDevices: for x509.UnknownAuthorityError sets ownedByOther to ownership status

### PR DESCRIPTION
**Pull request by jkralik**
* getDevice/getDevices: for UnknownAuthorityError sets ownedByOther to ownership status

### CODE REVIEW

File: client/getDevice_test.go [hunk 1]
```diff
@@ -287,3 +287,38 @@ func TestClientCheckForDuplicityDeviceInCache(t *testing.T) {
 	_, _, err = c.GetDevice(ctx, dev.DeviceID())
 	require.NoError(t, err)
 }
+
+func TestClientGetDeviceByIPOwnedByOther(t *testing.T) {
+	deviceID := test.MustFindDeviceByName(test.DevsimName)
+	ip4 := test.MustFindDeviceIP(test.DevsimName, test.IP4)
+
+	ctx, cancel := context.WithTimeout(context.Background(), TestTimeout*1000)
```
Comment (Danielius1922) [inline client/getDevice_test.go:295]: That's a long timeout. :D

File: client/client_test.go [hunk 2]
```diff
@@ -61,6 +65,38 @@ func (c *testSetupSecureClient) GetRootCertificateAuthorities() ([]*x509.Certifi
 }
 
 func NewTestSecureClient() (*client.Client, error) {
+	return newTestSecureClient(test.IdentityIntermediateCA, test.IdentityIntermediateCAKey)
+}
+
+func NewTestSecureClient1() (*client.Client, error) {
```
Comment (Danielius1922) [inline client/client_test.go:71]: I'm thinking about a more descriptive name than `NewTestSecureClient1`.  Maybe something like `NewTestSecureClientWithGeneratedCertificate`, though that's rather long.

**Review by Danielius1922 — APPROVED** [review]

File: client/client_test.go (hunk 2 above)
Comment (jkralik) [reply client/client_test.go]: Renamed

File: client/getDevice_test.go (hunk 1 above)
Comment (jkralik) [reply client/getDevice_test.go]: Fixed

**Pull request merged**